File: Python3/38. Count and Say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        ans = '1'
        for _ in range(n-1):
            count = 1
            pre = ans[0]
            temp = ''
            for i, c in enumerate(ans[1:], 1):
                if c == pre:
                    count += 1
                else:
                    temp += str(count) + pre
                    count = 1
                    pre = c
            temp += str(count) + pre
            ans = temp
            
        return ans
                

File: Java/38. Count and Say.java
class Solution {
    public String countAndSay(int n) {
        StringBuilder sb = new StringBuilder();
        sb.append('1');

        for (int i = 1; i < n; i++) {
            int count = 1;
            char pre = sb.charAt(0);
            StringBuilder temp = new StringBuilder();

            for (char c: sb.substring(1).toCharArray()) {
                if (c == pre) {
                    count++;
                } else {
                    temp.append(Integer.toString(count));
                    temp.append(pre);
                    pre = c;
                    count = 1;
                }
            }

            temp.append(Integer.toString(count));
            temp.append(pre);
            sb = temp;
        }

        return sb.toString();
    }
}